Util_UsageStatistics.php

```diff
@@ -1,218 +1,402 @@
+		if ( ! isset( $a[ $p1 ] ) ) {
 			return null;
+		}
 
-		$v = $a[$p1];
-		if ( is_null( $p2 ) )
+		$v = $a[ $p1 ];
+		if ( is_null( $p2 ) ) {
 			return $v;
-		if ( !isset( $v[$p2] ) )
+		}
+
+		if ( ! isset( $v[ $p2 ] ) ) {
 			return null;
+		}
 
-		$v = $v[$p2];
-		if ( is_null( $p3 ) )
+		$v = $v[ $p2 ];
+		if ( is_null( $p3 ) ) {
 			return $v;
-		if ( !isset( $v[$p3] ) )
+		}
+
+		if ( ! isset( $v[ $p3 ] ) ) {
 			return null;
+		}
 
-		return $v[$p3];
+		return $v[ $p3 ];
 	}
 
-
-
-	static public function v3( $a, $p ) {
-		if ( !is_array( $p ) ) {
+	/**
+	 * Retrieves a deeply nested value from a multidimensional array using an array of keys.
+	 *
+	 * This method safely navigates a nested array by following the sequence of keys provided.
+	 * If any key in the sequence is not set or the path is invalid, the method returns `null`.
+	 *
+	 * @param array            $a The input array to search.
+	 * @param string|int|array $p A single key or an array of keys representing the path to the desired value
+	 *                            If a single key is provided, it will be converted to an array.
+	 *
+	 * @return mixed|null The value at the specified key path, or null if the path is invalid or incomplete.
+	 */
+	public static function v3( $a, $p ) {
+		if ( ! is_array( $p ) ) {
 			$p = array( $p );
 		}
 
 		$actual = &$a;
-		for ( $i = 0; $i < count( $p ); $i++) {
-			$property = $p[$i];
+		$count  = count( $p );
+		for ( $i = 0; $i < $count; $i++ ) {
+			$property = $p[ $i ];
 
-			if ( !isset( $actual[$property] ) ) {
+			if ( ! isset( $actual[ $property ] ) ) {
 				return null;
 			}
 
-			$actual = &$actual[$property];
+			$actual = &$actual[ $property ];
 		}
 
 		return $actual;
 	}
 
-
-
-	static public function value_per_second( $a, $property1, $property2 ) {
-		if ( !isset( $a[$property1] ) || !isset( $a[$property2] ) )
+	/**
+	 * Calculates a value per second based on two properties of an array.
+	 *
+	 * This method checks the presence of two properties in an array and calculates the ratio
+	 * of the first property to the second, scaled by 100. If the properties are missing or
+	 * the second property is zero, a fallback value is returned.
+	 *
+	 * @param array      $a The input array containing the values to calculate.
+	 * @param string|int $property1 The key for the numerator value.
+	 * @param string|int $property2 The key for the denominator value.
+	 *
+	 * @return string The calculated value as a percentage (formatted to 1 decimal place), '0'
+	 *                if the denominator is zero, or 'n/a' if either key is missing.
+	 */
+	public static function value_per_second( $a, $property1, $property2 ) {
+		if ( ! isset( $a[ $property1 ] ) || ! isset( $a[ $property2 ] ) ) {
 			return 'n/a';
-		else if ( $a[$property2] == 0 )
-				return '0';
-			else
-				return sprintf( '%.1f', $a[$property1] / $a[$property2] * 100 );
+		} elseif ( 0 === $a[ $property2 ] ) {
```

Comment:
The value may not be an `int` -- change to `} elseif ( empty( $a[ $property2 ] ) ) {`.

Reply:
Updated